Sources/Core/AWSClientRuntime/UserAgent/FeatureMetadata.swift

```diff
@@ -0,0 +1,26 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+struct FeatureMetadata {
+    let feature: String
+    let version: String?
```

Comment:
In SEP, there's optional (0 to infinity) additional metadata appended to the end of the feature metadata. Is there a reason for the discrepancy?

Reply:
I think what @sichanyoo is suggesting here is that maybe we should have a class AdditionalMetadata from which subtypes of Metadata inherit. Then when we pass that metadata it should be an array of AdditionalMetadata since the specifications write `*(RWS additional-metadata)` which means we can have as much additional-metadata as we want in a row separated by the required white space

Reply:
I have added an `AdditionalMetadata` struct, which follows the SEP more closely.  Unlike the current `extras` dictionary, an array of `AdditionalMetadata` allows metadata with the same name to be transmitted more than once, which is not prohibited by the SEP.